modules/hostap/src/supp_api.c

```diff
@@ -1135,15 +1135,19 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 				goto out;
 			}
 
-			if (wpas_config_process_blob(wpa_s->conf, "ca_cert",
-					   enterprise_creds.ca_cert,
-					   enterprise_creds.ca_cert_len)) {
-				goto out;
-			}
+			if (false == ((params->security == WIFI_SECURITY_TYPE_EAP_PEAP_MSCHAPV2 ||
+			    params->security == WIFI_SECURITY_TYPE_EAP_TTLS_MSCHAPV2) &&
+			    (params->verify_peer_cert == 0))) {
+				if (wpas_config_process_blob(wpa_s->conf, "ca_cert",
+						   enterprise_creds.ca_cert,
+						   enterprise_creds.ca_cert_len)) {
+					goto out;
+				}
```

Comment:
The conditions are tough to glean, something like below might be better? Having said that, Peer cert verify is applicable to PEAP/TTLS too, as server always has the cert in those cases.

```
			if ((params->security != WIFI_SECURITY_TYPE_EAP_PEAP_MSCHAPV2 &&
				params->security != WIFI_SECURITY_TYPE_EAP_TTLS_MSCHAPV2) &&
				params->verify_peer_cert) {
				// Process the CA certificate blob
				if (wpas_config_process_blob(wpa_s->conf, "ca_cert",
									   enterprise_creds.ca_cert,
									   enterprise_creds.ca_cert_len)) {
				   // Handle error
				   // Add appropriate error handling code here
				}
			}
```

Reply:
> The conditions are tough to glean, something like below might be better? Having said that, Peer cert verify is applicable to PEAP/TTLS too, as server always has the cert in those cases.
> 
> ```
> 			if ((params->security != WIFI_SECURITY_TYPE_EAP_PEAP_MSCHAPV2 &&
> 				params->security != WIFI_SECURITY_TYPE_EAP_TTLS_MSCHAPV2) &&
> 				params->verify_peer_cert) {
> 				// Process the CA certificate blob
> 				if (wpas_config_process_blob(wpa_s->conf, "ca_cert",
> 									   enterprise_creds.ca_cert,
> 									   enterprise_creds.ca_cert_len)) {
> 				   // Handle error
> 				   // Add appropriate error handling code here
> 				}
> 			}
> ```

Hi @krish2718  the original purpose of @Qingling-Wu 's commit, is to support only check skipping CA certificate for  EAP-TTLS-MSCHAPV2 and EAP-PEAP-MSCHAPV2 case, but you suggested change is not the same logic.

Reply:
> AFAIK, the certificate validation is applicable for all enterprise methods including TTLS/PEAP, so, we should just do this check for all enterprise methods.

Hi @krish2718,
Yes, sure, the certificate verify is a must for all applicable enterprise methods including TTLS/PEAP except for EAP_PEAP_MSCHAPV2 and EAP_TTLS_MSCHAPV2.

For enterprise methods except for EAP_PEAP_MSCHAPV2 and EAP_TTLS_MSCHAPV2, blow check is always true, params->verify_peer_cert is ignored, and certificate would be verified.
Only for EAP_PEAP_MSCHAPV2 or EAP_TTLS_MSCHAPV2, params->verify_peer_cert would be checked. And if params->verify_peer_cert == 0, certificate would be skipped, if params->verify_peer_cert == 1, certificate would be verified.
((params->security == WIFI_SECURITY_TYPE_EAP_PEAP_MSCHAPV2 ||
			    params->security == WIFI_SECURITY_TYPE_EAP_TTLS_MSCHAPV2) &&
			    (params->verify_peer_cert == 0))

Reply:
> except for EAP_PEAP_MSCHAPV2 and EAP_TTLS_MSCHAPV2.

This is the point where I am clear, AFAIK, even when using MSCHAPv2 the client can still verify the server certificate using CA, no?

Reply:
> > except for EAP_PEAP_MSCHAPV2 and EAP_TTLS_MSCHAPV2.
> 
> This is the point where I am not clear, AFAIK, even when using MSCHAPv2 the client can still verify the server certificate using CA, no?

Yes, for EAP_PEAP_MSCHAPV2 and EAP_TTLS_MSCHAPV2, certificate verify is optional, the client can verify the server certificate using CA, could specify whether use CA or not by setting ‘-A <0/1> ’.